### PR TITLE
Ft signup #164691440

### PR DESCRIPTION
**Pull request by Busobozihakiim**
#### What does this PR do?
- Enables users to create accounts they will use for logging in.

#### Description of Task to be completed?
- Write the /signup/endpoint to implement the feature 
- Write helper function that checks if data input by a user is valid  and creates the user
- Add method to the user modal class that creates a user and adds it to a list

#### How should this be manually tested?
- https://epicmail007.herokuapp.com/api/v1/auth/signup

 #### Relevant Pivotal tracker Id
#164691440

### CODE REVIEW

**Comment by kenneth051** [comment]
@Busobozihakiim , could you look into the failing Travis build.?

File: api/v1/models/user_model.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+"""USER ENTITY """
+import uuid
+from werkzeug.security import generate_password_hash
+USERS = []
```
Comment (kenneth051) [inline api/v1/models/user_model.py]: @Busobozihakiim , we should make this(USER =[]) a class variable.
 read more about class variables, please. right now it is a global variable. we should try to ensure everything is OOP

File: api/v1/models/user_model.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class Users:
```
Comment (kenneth051) [inline api/v1/models/user_model.py:6]: @Busobozihakiim,  nice work he, however, The user class should have all the methods that deal with a user, for instance, creating a user, getting a user or deleting a user. we should put all the validation in a validation file so we just call them in other files when we need to use them

**Review by kenneth051 — CHANGES_REQUESTED** [review]

File: api/v1/models/user_model.py (hunk 2 above)
Comment (Busobozihakiim) [reply api/v1/models/user_model.py:6]: the validation file does exist, and in this Pr, the user class currently  has methods of creating a user and checking if  the email they use for login exists

Comment (kenneth051) [reply api/v1/models/user_model.py:6]: @Busobozihakiim , my mistake..I was looking at someone else's PR..

File: api/v1/models/user_model.py (hunk 1 above)
Comment (Busobozihakiim) [reply api/v1/models/user_model.py]: This has been implemeted

**Pull request merged**